NPC/kode-frieze/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+extern crate wasmdome_mech_sdk as mech;
+
+use mech::*;
+
+mech_handler!(handler);
+
+pub fn handler(mech: impl MechInstruments) -> Vec<MechCommand> {
+     if !is_in_center(&mech) {
+          let mut cmds = move_to_center(&mech);
+          cmds.push(mech.request_radar());
+          cmds
+     } else if let Some(scan) = mech.last_radar_scan() {
+          seek_and_destroy_target(&mech, scan)
+     } else {
+          vec![mech.request_radar(), mech.register_acc(ECX, 1)]
+     }
+}
+
+/// We accept being 1 square away to cover the even board width and height edge case
+fn is_in_center(mech: &impl MechInstruments) -> bool {
+     if distance_to(&mech.position(), &compute_center(mech.world_size())) <= 1 {
+          true
+     } else {
+          false
+     }
+}
+
+// Returns None if target already exists in the register
+// Returns a mech command to set the target otherwise
+fn seek_and_destroy_target(mech: &impl MechInstruments, scan: Vec<RadarPing>) -> Vec<MechCommand> {
+     let scan = scan
+          .iter()
+          .filter(|s| distance_to(&mech.position(), &s.location) <= mech.primary_range())
+          .filter(|s| s.foe)
+          .collect::<Vec<_>>();
+
+     match mech.register_get(EBX) {
+          Some(RegisterValue::Text(target)) => {
+               if let Some(target) = scan
+                    .iter()
+                    .filter(|s| s.id == *target)
+                    .collect::<Vec<_>>()
+                    .get(0)
+               {
+                    vec![
```

Comment:
`register_set` doesn't cost APs... so you can still fire the primary twice on the same round you set the register.